mirage/seed_image/psf_generator.py

```diff
@@ -0,0 +1,160 @@
+#! /suyr/bin/env python
+
+"""
+Class to generate PSFs to be used with Mirage
+"""
+import os
+
+import numpy as np
+from astropy.io import fits
+from photutils.psf import FittableImageModel
+
+class PSF():
+    def __init__(self, x_position, y_position, psf_base, interval=0.25, oversampling=1):
+        """
+        Populate FittableImageModel instance with data from the appropriate PSF file
+
+        Parameters:
+        -----------
+        x_position : float
+            Column number on the detector where the PSF will be located
+
+        y_position : float
+            Row number on the detector where the PSF will be located
+
+        psf_base : str
+            Base of the PSF filename (e.g. "nircam"
+
+        interval : float
+            Sub-pixel centering resolution of the PSF library (e.g. 0.25)
+
+        oversampling : int
```

Comment:
Could `oversampling` and `interval` be read from the PSF file header, rather than explicitly provided as parameters?

Reply:
`interval` is not in the file header. It's essentially the resolution of the files in the library. This function was written as a quick fix under the assumption that we would merge the branch and use `fittableImageModel` with the current PSF libraries. But now that we have @shanosborne's tool to create new libraries, I was thinking that development in this branch could push forward to use those libraries and get the position-dependent PSFs implemented. In that case, this class will have to be updated.